lib/teamsnap.rb

```diff
@@ -137,13 +138,40 @@ def init(opts = {})
       collection = TeamSnap.run_init(:get, "/", {})
 
       classes = []
-      client.in_parallel do
-        classes = collection
-          .fetch(:links)
-          .map { |link| classify_rel(link) }
+
+      schema = collection
+        .fetch(:links) {{}}
+        .detect { |link| link[:rel] == "schemas" }
+
+      if schema && schema[:href]
+        hrefToRel = collection
+          .fetch(:links, {})
+          .map { |link| [link[:href], link[:rel]]}
+          .reject { |href_rel| EXCLUDED_RELS.include?(href_rel[1]) }
+          .to_h
+
+        resp = client.get(schema[:href])
+        if resp.status == 200
+          collections = Oj.load(resp.body)
+          classes = collections.map { |collection|
+            col = collection.fetch(:collection) {{}}
+            if rel = hrefToRel[col[:href]]
+              create_collection_class(rel, col[:href], nil, col)
+            end
+          }
           .compact
+        else
+          error_message = TeamSnap.parse_error(resp)
+          raise TeamSnap::Error.new(error_message)
+        end
+      else
+        client.in_parallel do
+          classes = collection
+            .fetch(:links)
```

Comment:
no default here, I think most endpoints should have `:links` but might want to guard against that?